pkg/asset/store.go

```diff
@@ -16,45 +17,100 @@ type Store interface {
 
 // StoreImpl is the implementation of Store.
 type StoreImpl struct {
+	fileFetcher *fileFetcher
+	// so that each asset can load and reuse them.
 	assets map[reflect.Type]Asset
+	// dirtyAssets marks whether or not the asset is dirty.
+	dirtyAssets map[reflect.Type]bool
```

Comment:
I still feel pretty strongly that the asset and the dirty flag should be stored in the same map. They are always both fetched and stored at the same time.

Reply:
Yeah +1, 
i think we should change `assets map[reflect.Type]Assets` to something like  `assets ma[reflect.Type]struct {dirty, Asset}`